smac/intensification/intensification.py

```diff
@@ -137,30 +192,303 @@ def __init__(self, tae_runner: ExecuteTARun,
         self.inc_sum_cost = np.inf
         self.N = -1
 
-    def eval_challenger(self,
+    def get_next_run(self,
+                     challengers: typing.Optional[typing.List[Configuration]],
+                     incumbent: Configuration,
+                     chooser: typing.Optional[EPMChooser],
+                     run_history: RunHistory,
+                     repeat_configs: bool = True,
+                     ) -> typing.Tuple[IntensifierBehest, RunInfo]:
+        """
+        This procedure is in charge of generating a RunInfo object to comply
+        with lines 7 (in case stage is stage==RUN_INCUMBENT) or line 12
+        (In case of stage==RUN_CHALLENGER)
+
+        A RunInfo object encapsulates the necessary information for a worker
+        to execute the job, nevertheless, a challenger is not always available.
+        This could happen because no more configurations are available or the new
+        configuration to try was already executed.
+
+        To circumvent this, a behest is also returned:
+
+        - (behest=RUN) Run the RunInfo object (Normal Execution
+        - (behest=TERMINATE) No more challengers available (Line 19 should occur)
+        - (behest=SKIP) Skip this iteration. No challenger is available, in particular
+            because challenger is the same as incumbent
+
+
+        Parameters
+        ----------
+        challengers : typing.List[Configuration]
+            promising configurations
+        incumbent: Configuration
+            incumbent configuration
+        chooser : smac.optimizer.epm_configuration_chooser.EPMChooser
+            optimizer that generates next configurations to use for racing
+        run_history : typing.Optional[smac.runhistory.runhistory.RunHistory]
+            stores all runs we ran so far
+        repeat_configs : bool
+            if False, an evaluated configuration will not be generated again
+
+        Returns
+        -------
+        behest: IntensifierBehest
+            What should the smbo object do with the runinfo.
+        run_info: RunInfo
+            An object that encapsulates necessary information for a config run
+        """
+
+        # In case a crash happens, and FirstRunCrashedException prevents a
+        # failure, revert back to running the incumbent
+        # Challenger case is by construction ok, as there is no special
+        # stage for its processing
+        if self.stage == IntensifierStage.PROCESS_FIRST_CONFIG_RUN:
+            self.stage = IntensifierStage.RUN_FIRST_CONFIG
+        elif self.stage == IntensifierStage.PROCESS_INCUMBENT_RUN:
+            self.stage = IntensifierStage.RUN_INCUMBENT
+
+        try:
+
+            # Line 2:
+            # Understand who is the challenger.
+            # Not necessarily a challenger is always a new configuration,
```

Comment:
Could you please write 'A challenger is not necessarily a new configuration'?

Reply:
Done!